File: src/test/java/net/dontdrinkandroot/gitki/model/SimpleGitUser.java
package net.dontdrinkandroot.gitki.model;

import java.util.Objects;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class SimpleGitUser implements GitUser
{
    private String email;

    private String fullname;

    public SimpleGitUser(String fullname, String email)
    {
        Objects.requireNonNull(fullname);
        Objects.requireNonNull(email);
        this.email = email;
        this.fullname = fullname;
    }

    @Override
    public String getFullName()
    {
        return null;
    }

    @Override
    public String getEmail()
    {
        return this.email;
    }

    @Override
    public boolean equals(Object o)
    {
        if (this == o)
            return true;
        if (o == null || this.getClass() != o.getClass())
            return false;

        SimpleGitUser that = (SimpleGitUser) o;

        return this.email.equals(that.email);
    }

    @Override
    public int hashCode()
    {
        return this.email.hashCode();
    }
}

File: src/test/kotlin/net/dontdrinkandroot/gitki/model/SimpleGitUser.kt
package net.dontdrinkandroot.gitki.model

data class SimpleGitUser(override val fullName: String, override val email: String) : GitUser {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || this.javaClass != other.javaClass) return false
        val that = other as SimpleGitUser
        return email == that.email
    }

    override fun hashCode(): Int = email.hashCode()
}